addons/purchase/models/purchase.py

```diff
@@ -1010,6 +1010,14 @@ def make_po(self):
                 self.env['purchase.order.line'].create(vals)
         return res
 
+    @api.multi
+    def open_purchase_order(self):
+        action = self.env.ref('purchase.action_open_purchase_from_procurement')
+        action_dict = action.read()[0]
+        action_dict['res_id'] = self.purchase_id.id
+        action_dict['target'] = 'current'
```

Comment:
``current`` is the default target, and since this is the only consumer of the action, if you want to be explicit it can be set explicitly there.